src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,68 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.consts.ErrorMessage;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.Section;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.List;
+
+public class PathFinder {
+    private DijkstraShortestPath<Station, DefaultWeightedEdge> dijkstraShortestPath;
+
+    public PathFinder(List<Line> lines) {
+        WeightedMultigraph<Station, DefaultWeightedEdge> graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
+        registerLines(graph, lines);
+        this.dijkstraShortestPath = new DijkstraShortestPath(graph);
+    }
+
+    public static PathFinder from(List<Line> lines) {
+        return new PathFinder(lines);
+    }
+
+    private void registerLines(WeightedMultigraph<Station, DefaultWeightedEdge> routeGraph, List<Line> lines) {
+        for (Line line : lines) {
+            addVertexes(routeGraph, line);
+            addEdges(routeGraph, line);
+        }
+    }
+
+    private void addVertexes(WeightedMultigraph<Station, DefaultWeightedEdge> routeGraph, Line line) {
+        line.getStations().stream()
+                .forEach(station -> routeGraph.addVertex(station));
```

Comment:
stream()과 forEach()를 같이 안써도 될 것 같아요 😄 
```suggestion
       line.getStations()
             .forEach(routeGraph::addVertex);
```

Reply:
앗 그러네요 😅  감사합니다!!